### PR TITLE
Feature/58725 webchat v3 settings

### PR DESCRIPTION
**Pull request by dshire**
## References
User Story https://cognigy.visualstudio.com/Cognigy.AI/_workitems/edit/58725

This PR implements the new v3 settings following the specification from spec https://cognigy.visualstudio.com/Engineering/_git/technical-design-documents/pullrequest/24778

## Approach 
The original approach that we keep the old settings in place (see US) and rename the new ones was changed to fully using the new settings and their structure. This way we reduce future overhead for each new setting.

There is currently no "translation" from the old settings to the new, so you'll have to set the new settings if you want to change anything from default.

## Review
For the review, please do some functional testing of the different webchat features and check the changes for any obvious issues.
If you see anything missing that is not mentioned in the TO DOs below, please also mention it in your PR comments.

## TO DOs
Some settings are not yet implemented, as they basically are new features. They will be added in future smaller PRs as iterations to this one.

#### New functionality
- Implement "useOtherAgentLogo" setting: Use different logo and name for Virtual & Human Agent Messages
- Implement new disableBranding functionality allowing for custom branding text (now called layout.watermark / layout.watermarkText)
- implement colors being loaded from settings (currently only primaryColor is loaded)

#### Other TO DOs
- use rating.chatStatusMessage in RatingConfirmation.tsx
- ~~use maintenance.title~~

#### Blocked
- ~~privacy notice finalization (missing definitions for privacy link button, see spec discussion)~~
- chat options settings (TTS not yet implemented / ~~quickreplybuttons interface from spec not fitting~~)
- ~~homescreen conversation starters & teaser message conversation starters (unclear in settings structure, see spec)~~

### CODE REVIEW

**Comment by sushmi21** [comment]
The setting `useOtherAgentLogo` is not used anywhere. According to the word document linked in the spec, if this setting is false, we should use layout.title and layout.logoUrl for bot and live agent messages. Else we should use layout.botAvatarName, layout.botLogoUrl for messages from bot and layout.agentAvatarName, layout.agentLogoUrl for messages from live agent.

**Comment by dshire** [comment]
> The setting `useOtherAgentLogo` is not used anywhere. According to the word document linked in the spec, if this setting is false, we should use `layout.title` and `layout.logoUrl` for bot and live agent messages. Else we should use `layout.botAvatarName`, `layout.botLogoUr ` for messages from bot and `layout.agentAvatarName`, `layout.agentLogoUrl` for messages from live agent.

yes, this is new functionality, it's mentioned above in the PR description and will be handled in a follow-up for these features

**Comment by sushmi21** [comment]
> agentAvatarName

Oops. Sorry I missed.

File: src/webchat-ui/components/plugins/input/base/BaseInput.tsx [hunk 1]
```diff
@@ -391,12 +393,20 @@ export class BaseInput extends React.PureComponent<IBaseInputProps, IBaseInputSt
 		const { text, textActive, speechResult: speechInterim } = state;
 
 		const {
-			disableInputAutocomplete,
-			disableInputAutofocus,
+			layout,
 			enableFileAttachment,
```
Comment (sushmi21) [inline src/webchat-ui/components/plugins/input/base/BaseInput.tsx:397]: This is different now `fileStorageSettings.enabled`. We also need to handle the new setting `fileStorageSettings.dropzoneText`

**Comment by sushmi21** [comment]
`src/webchat/helper/businessHours.ts` In this file, we need to modify `businessHours.businessHours` to `businessHours.times`

File: src/webchat-ui/components/plugins/input/base/BaseInput.tsx (hunk 1 above)
Comment (sushmi21) [reply src/webchat-ui/components/plugins/input/base/BaseInput.tsx:397]: Also `businessHours.timeZone `should be changed to` businessHours.timeyone`

File: src/webchat-ui/components/WebchatUI.tsx [hunk 2]
```diff
@@ -948,37 +947,43 @@ export class WebchatUI extends React.PureComponent<
...
-				return "Previous conversations";
+				return config.settings.homeScreen.previousConversations.title || "Previous conversations";
 			}
 			if (showChatOptionsScreen) {
-				return "Chat options";
+				return config.settings.chatOptions.title || "Chat options";
 			}
 			if (showRatingScreen) {
-				return "Conversation rating";
+				return config.settings.chatOptions.rating.title || "Conversation rating";
```
[the remark this answers is not in the document]
Comment (sushmi21) [reply src/webchat-ui/components/WebchatUI.tsx:960]: I already wrote a comment in spec

**Comment by sushmi21** [comment]
> `src/webchat/helper/businessHours.ts` In this file, we need to modify `businessHours.businessHours` to `businessHours.times`

Also `businessHours.timeZone `should be changed to` businessHours.timeyone`

File: src/webchat-ui/components/presentational/chat-options/ChatOptionsFooter.tsx [hunk 3]
```diff
@@ -29,13 +30,20 @@ const StyledFooterTypography = styled(Typography)(() => ({
...
+	const { settings } = props;
+	const { chatOptions } = settings;
+
+	const footerItem1Text = chatOptions?.footer?.items[0]?.title || "Imprint";
+	const footerItem2Text = chatOptions?.footer?.items[1]?.title || "Data Privacy";
 
-	const footerItem1URL = "https://www.cognigy.com/legal-notice";
-	const footerItem2URL = "https://www.cognigy.com/privacy-policy";
+	const footerItem1URL = chatOptions?.footer?.items[0]?.url || "https://www.cognigy.com/legal-notice";
+	const footerItem2URL = chatOptions?.footer?.items[1]?.url || "https://www.cognigy.com/privacy-policy";
```
Comment (sushmi21) [inline src/webchat-ui/components/presentational/chat-options/ChatOptionsFooter.tsx:47]: Can you please add a condition (chatOptions?.footer?.items[1])  in line 60 to display the second link?

**Comment by sushmi21** [comment]
We should use `rating.chatStatusMessage` in `RatingConfirmation.tsx`

**Comment by sushmi21** [comment]
I think we have a new TODO to use `maintenance.title`

File: src/webchat-ui/components/plugins/input/base/BaseInput.tsx (hunk 1 above)
Comment (dshire) [reply src/webchat-ui/components/plugins/input/base/BaseInput.tsx:397]: hm there was enableFileUpload and enableFileAttachment, not sure what the difference is but the spec says to replace enableFileUpload. I think we need to clarify this.

**Comment by dshire** [comment]
added both rating.chatStatusMessage and maintenance.title to ToDos

File: src/webchat-ui/components/plugins/input/base/BaseInput.tsx (hunk 1 above)
Comment (dshire) [reply src/webchat-ui/components/plugins/input/base/BaseInput.tsx]: after investigating, I removed both old settings mentioned above and implemented it to use the two filestoragesettings keys

**Comment by sushmi21** [comment]
I do not see the default header logo in the webchat. Does it work for you?

**Comment by sushmi21** [comment]
BusinessHours feature is boken. Not because of this PR though. We should handle it in a separate PR

File: src/webchat-ui/components/presentational/HomeScreen.tsx [hunk 4]
```diff
@@ -141,31 +144,11 @@ export const HomeScreen: React.FC<IHomeScreenProps> = props => {
 		onStartConversation,
 	} = props;
 
-	// TODO: Load buttons from new endpoint config
-	const buttons: IWebchatButton[] = [
-		{
-			title: "Cognigy Homepage",
-			type: "web_url",
```
Comment (sushmi21) [inline src/webchat-ui/components/presentational/HomeScreen.tsx:148]: According to the spec, the buttons will be either of type postback or 'url'. ActionButton component handles only links of type 'web_url'. So, when I add a conversation starter url according to the spec, it does not show up on the UI. We need to do a type conversion here or handle url type in chat-components

Comment (sushmi21) [reply src/webchat-ui/components/presentational/HomeScreen.tsx:148]: This is same in Chat options screen and in the teaser message

**Comment by sushmi21** [comment]
Rest looks good to me.

**Pull request merged**